src/blueprint/server.rs

```diff
@@ -4,10 +4,31 @@ use std::net::{AddrParseError, IpAddr};
 use derive_setters::Setters;
 use hyper::header::{HeaderName, HeaderValue};
 use hyper::HeaderMap;
+use serde::{Deserialize, Serialize};
 
 use crate::config;
 use crate::valid::{Valid, ValidationError};
 
+#[derive(Deserialize, Serialize, Clone, Debug, Setters)]
+#[serde(rename_all = "camelCase", default)]
+pub struct Http {
+  pub version: HttpVersion,
+  pub cert_path: Option<String>,
+  pub key_path: Option<String>,
+}
+
+impl Default for Http {
+  fn default() -> Self {
+    Http { version: HttpVersion::HTTP1, cert_path: None, key_path: None }
+  }
+}
+
+#[derive(Deserialize, Serialize, Debug, PartialEq, Eq, Clone)]
+pub enum HttpVersion {
+  HTTP1,
+  HTTP2,
```

Comment:
The blueprint::server should have more precise semantics. Always remember - _A BLUEPRINT CAN NEVER BE INCOMPLETE_.

```suggestion
  HTTP2 {cert: String, key: String},
```